src/pages/workspace/WorkspaceSettingsPage.js

```diff
@@ -39,125 +39,116 @@ const defaultProps = {
     ...policyDefaultProps,
 };
 
-class WorkspaceSettingsPage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.getCurrencyItems = this.getCurrencyItems.bind(this);
-        this.submit = this.submit.bind(this);
-        this.validate = this.validate.bind(this);
-    }
-
+function WorkspaceSettingsPage(props) {
     /**
      * @returns {Object[]}
      */
-    getCurrencyItems() {
-        const currencyListKeys = _.keys(this.props.currencyList);
+    function getCurrencyItems() {
+        const currencyListKeys = _.keys(props.currencyList);
         return _.map(currencyListKeys, currencyCode => ({
             value: currencyCode,
-            label: `${currencyCode} - ${this.props.currencyList[currencyCode].symbol}`,
+            label: `${currencyCode} - ${props.currencyList[currencyCode].symbol}`,
         }));
     }
 
-    submit(values) {
-        if (this.props.policy.isPolicyUpdating) {
+    const submit = useCallback((values) => {
+        if (props.policy.isPolicyUpdating) {
             return;
         }
         const outputCurrency = values.currency;
-        Policy.updateGeneralSettings(this.props.policy.id, values.name, outputCurrency);
+        Policy.updateGeneralSettings(props.policy.id, values.name, outputCurrency);
         Keyboard.dismiss();
-        Navigation.navigate(ROUTES.getWorkspaceInitialRoute(this.props.policy.id));
-    }
+        Navigation.navigate(ROUTES.getWorkspaceInitialRoute(props.policy.id));
+    }, [props.policy.id, props.policy.isPolicyUpdating]);
 
-    validate(values) {
+    const validate = useCallback((values) => {
         const errors = {};
         const name = values.name.trim();
 
         if (!name || !name.length) {
-            errors.name = this.props.translate('workspace.editor.nameIsRequiredError');
+            errors.name = props.translate('workspace.editor.nameIsRequiredError');
         }
 
         return errors;
-    }
+    }, [props]);
```

Comment:
I think this has to have `props` as a dependency, because I don't think it can just have `props.translate`. Is there a way around this?

Reply:
@dangrous In this [PR](https://github.com/Expensify/App/pull/17473/files#diff-9610997c7daea382e6bf987853087fb30a336656326a938e7c0ced230f8a40d4R32), we have defined a const for translation and have it as dependency.

Something like:
`const nameIsRequiredError =  props.translate('workspace.editor.nameIsRequiredError')`

Reply:
oh neat, I can update that - thank you for the reference!